### PR TITLE
[Bug] Fixes spaces in variable resolution for base parser

### LINKED ISSUE #56: Replace variables doesn't work well when there is a space
!if $ (SECURE_BOOT_ENABLE) == TRUE
becomes 
0!if $ (SECURE_BOOT_ENABLE) == TRUE
Which means we don't resolve it as a conditional

### PR DESCRIPTION
**Pull request by matthewfcarlson**
fixes #56

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/base_parser_test.py [hunk 1]
```diff
@@ -39,6 +39,16 @@ def test_replace_macro_using_dollarsign(self):
...
+    
+    def test_replace_macro_using_dollarsign_spaces(self):
+        parser = BaseParser("")
+        parser.SetInputVars({
+            "name": "sean"
+        })
+        line = "Hello $ (name)!"
+        self.assertEqual(parser.ReplaceVariables(line), "Hello sean!")
+        line = "Hello $( name )!"
+        self.assertEqual(parser.ReplaceVariables(line), "Hello sean!")
```
Comment (corthon) [inline edk2toollib/uefi/edk2/parsers/base_parser_test.py]: Can you add a test for the negative case? e.g. missing $

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: like `line = "Hello name"` or `line = "Hello $name"`?

Comment (corthon) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: First one, then the other! :)

**Comment by spbrogan** [comment]
Is this syntax valid according to DSC spec?

**Comment by matthewfcarlson** [comment]
according to the spec here, no. https://edk2-docs.gitbooks.io/edk-ii-dsc-specification/content/v/release/1.28/3_edk_ii_dsc_file_format/33_platform_dsc_definition.html but then here: https://edk2-docs.gitbooks.io/edk-ii-dsc-specification/content/v/release/1.28/appendix_b_sample_edk_ii_dsc_file.html they use $ () in a few places. I say let's go with the spec on this one.

**Comment by Javagedes** [comment]
Closing due to the conversation in #56. If this is incorrect, feel free to re-open.
